typescript/sdk/src/metadata/ChainMetadataManager.ts

```diff
@@ -56,59 +62,56 @@ export class ChainMetadataManager<MetaExt = {}> {
 
   /**
    * Add a chain to the MultiProvider
-   * @throws if chain's name or domain/chain ID collide
+   * @throws if chain's name or domain ID collide
    */
   addChain(metadata: ChainMetadata<MetaExt>): void {
     ChainMetadataSchema.parse(metadata);
-    // Ensure no two chains have overlapping names/domainIds/chainIds
+    // Ensure no two chains have overlapping names/domainIds
     for (const chainMetadata of Object.values(this.metadata)) {
-      const { name, chainId, domainId } = chainMetadata;
+      const { name, domainId } = chainMetadata;
       if (name == metadata.name)
         throw new Error(`Duplicate chain name: ${name}`);
-      // Chain and Domain Ids should be globally unique
```

Comment:
CLI still defaults to domainId being the chainId, so we'll still be able to catch duplicates in the general case - we can also add stricter tests in the registry itself rather than trying to shape it here

> don't create new deploys on chains that already have mailboxes

the cli currently checks if we've got a mailbox for the selected `chain`